### PR TITLE
enhancement(lua transform): Upgrade Lua runtime to 5.4

### LINKED ISSUE #6638: Upgrade Lua runtime to 5.4
Requested by user in discord: https://discord.com/channels/742820443487993987/746070591097798688/817387510811721738

### PR DESCRIPTION
**Pull request by pablosichert**
Closes #6638.

### CODE REVIEW

File: src/transforms/lua/v2/interop/util.rs [hunk 1]
```diff
@@ -77,3 +77,19 @@ where
...
+        LuaValue::LightUserData(_) => "light userdata",
+        LuaValue::Integer(_) => "integer",
+        LuaValue::Number(_) => "number",
+        LuaValue::String(_) => "string",
+        LuaValue::Table(_) => "table",
+        LuaValue::Function(_) => "function",
+        LuaValue::Thread(_) => "thread",
+        LuaValue::UserData(_) | LuaValue::Error(_) => "userdata",
+    }
+}
```
Comment (pablosichert) [inline src/transforms/lua/v2/interop/util.rs:95]: It's a bit weird that they publicly expose the [`Error`](https://github.com/amethyst/rlua/blob/v0.17.0/src/error.rs#L63-L71) type with a `from` field taking `&'static str`, but not the [`type_name`](https://github.com/amethyst/rlua/blob/v0.17.0/src/value.rs#L50-L63) implementation.

**Review by ktff — APPROVED** [review]
LGTM (the failing check seems to be unrelated)

File: src/transforms/lua/v2/interop/util.rs (hunk 1 above)
Comment (jszwedko) [reply src/transforms/lua/v2/interop/util.rs:95]: Hmm, that is odd. Maybe worth an issue on the crate?

**Review by jszwedko — APPROVED** [review]
Looks good!

**Pull request merged**